01-js/hard/calculator.js

```diff
@@ -16,6 +16,48 @@
   Once you've implemented the logic, test your code by running
 */
 
-class Calculator {}
+class Calculator {
+  constructor() {
+    this.total=0
+}
+
+add(number) {
+  this.total+=number
+}
+
+subtract(number) {
+  this.total-=number
+}
+
+multiply(number) {
+  this.total*=number
+}
+
+divide(number) {
+  if(number===0)
+  {
+    throw new Error("Error")
+  }
+  this.total/=number
+}
+
+clear() {
+  this.total=0
+}
+
+getResult() {
+  return this.total
+}
+
+calculate(expression) {
+  let expOutput=eval(expression)
+  if(expOutput === Infinity)
+  {
+    throw new Error("Error")
+  }
+  this.total=expOutput
+  return this.total
+}
```

Comment:
@lasya-codes I don't see this approach passing the test cases locally. Is there something that I am missing?

Reply:
Hi @Gowtham-Enlume , Thank you for reviewing the code.  It is passing all the test cases as shown in the below image. I did try checking manually all the cases including 
<img width="535" alt="Week1_Review" src="https://github.com/100xdevs-cohort-2/assignments/assets/166899754/5d537dd4-ec36-4761-92b3-692d305d9e91">